### ISSUE TITLE
An option to mark duplicates

### ISSUE THREAD

**Issue by indus** [mapbox/tippecanoe#353]
Probably I'm facing a very individual case here, but I have the need to actually split polygones on tile edges but also have to avoid duplicates when I call [querySourceFeatures](https://www.mapbox.com/mapbox-gl-js/api/#Map#querySourceFeatures) with mapbox-gl-js.
So for me it would be nice to have an option to mark them with a flag in properties (e.g.  ```"duplicate":true```) to filter them later on when making the query. Maybe it is not a big thing to keep one "as is" and write the flag only to any further duplicate that gets created while splitting?!

**Comment by ericfischer** [comment]
Hmm, I'll have to think about how I could do this. It's tricky because clipping doesn't happen until after splitting, so the "central" tile of a polygon might turn out to be in a hole in it, causing that polygon not to appear at all in the tile that is supposedly its primary appearance.

I think the best thing you could do at the moment would be to give each feature a unique ID and then check the results from `querySourceFeatures` to see if you got two features back with the same ID.

**Comment by indus** [comment]
Right, like this example: https://www.mapbox.com/mapbox-gl-js/example/filter-features-within-map-view/
Because it is a big dataset I just wanted to avoid the second loop/filter on all features. But maybe a customized querySourceFeatures let me do that as well...
